clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -211,17 +214,24 @@ public RecordAppendResult append(TopicPartition tp,
         }
     }
 
+    private MemoryRecordsBuilder getRecordsBuilder(ByteBuffer buffer) {
+        if (transactionState != null) {
+            return MemoryRecords.builder(buffer, LogEntry.MAGIC_VALUE_V2, compression, TimestampType.CREATE_TIME,
```

Comment:
I think we need to use a version which lets us pass the `batchSize` as we do below.